src/pbkdf2.rs

```diff
@@ -150,14 +148,39 @@ pub fn derive(prf: &'static PRF, iterations: usize, salt: &[u8], secret: &[u8],
     // Clear |out|.
     polyfill::slice::fill(out, 0);
 
+    let mut idx: u32 = 0;
+
+    for chunk in out.chunks_mut(output_len) {
+        idx += 1;
+        let chunk_len = chunk.len();
+
+        if chunk_len == output_len {
+            derive_block(&secret, iterations, salt, idx, output_len, chunk);
+        } else {
+            assert!(output_len <= 64);
+            let mut tmp = [0u8; 64];
+            derive_block(&secret, iterations, salt, idx, output_len, &mut tmp);
+            chunk.copy_from_slice(&tmp[..chunk_len]);
+        }
+    }
+}
+
+fn derive_block(
+        secret: &hmac::SigningKey,
+        iterations: usize,
+        salt: &[u8],
+        idx: u32,
+        len: usize,
```

Comment:
I think that `len` is redundant with `out.len()`. We should just use `out.len()`.